src/commons/asBaseComponent.tsx

```diff
@@ -25,17 +25,30 @@ function asBaseComponent<PROPS, STATICS = {}>(WrappedComponent: React.ComponentT
     static propTypes: any;
     static defaultProps: any;
 
+    state = {
+      error: false
+    };
+
     static getThemeProps = (props: any, context: any) => {
       return Modifiers.getThemeProps.call(WrappedComponent, props, context);
     }
 
+    static getDerivedStateFromError(error: any) {
+      UIComponent.defaultProps.onError(error, WrappedComponent.defaultProps);
+      return {error: true};
+    }
+
     render() {
       const themeProps = BaseComponent.getThemeProps(this.props, this.context);
       const modifiers = Modifiers.generateModifiersStyle(undefined, themeProps);
       // TODO: omit original modifiers props (left, right, flex, etc..)
       // Because they throws an error when being passed to RNView on Android
       const {forwardedRef, ...others} = themeProps;
-      return <WrappedComponent {...others} modifiers={modifiers} ref={forwardedRef}/>;
+      return this.state.error ? (
+        UIComponent.defaultProps.renderError
```

Comment:
Same here, better check how it works in public..

Reply:
Fixed :)